ghost/mailgun-client/lib/mailgun-client.js

```diff
@@ -203,6 +203,12 @@ module.exports = class MailgunClient {
     normalizeEvent(event) {
         const providerId = event?.message?.headers['message-id'];
 
+        if (!providerId && !(event['user-variables'] && event['user-variables']['email-id'])) {
```

Comment:
Double checking the use of `&&` here in this condition, what happens if we have a case where `providerId` is empty but `event['user-variables']['email-id']` exists ? If such a case does happen, we'll go ahead with this as a valid event and is that acceptable ?

Reply:
Yes, that is acceptable because we can still transform the providerId into an emailId later: https://github.com/TryGhost/Ghost/blob/aa693039a354a186f1ecd4a279a7b015f67bddc7/ghost/email-service/lib/email-event-processor.js#L147-L168

I don't know if we ever get those events, but we have been supporting them in the past

Reply:
Ah cool, all good then 😄